app/lib/actions/middleware/execute_if_contents_changed.rb

```diff
@@ -12,12 +12,17 @@ def finalize(*args)
       private
 
       def execute?
-        if action.input.keys.include?('contents_changed') && !action.input['contents_changed'] && !Setting[:force_post_sync_actions]
-          self.action.output[:post_sync_skipped] = true
-          false
-        else
-          true
+        execute = Setting[:force_post_sync_actions]
+        execute ||= action.input.keys.include?('contents_changed') && action.input['contents_changed']
+        repo_id = action.input['id'] || action.input['repo_id']
+
+        if !execute && repo_id
+          repo = ::Katello::Repository.find(repo_id)
+          execute = repo.pulp_counts_differ? if repo
         end
+
+        self.action.output[:post_sync_skipped] = true unless execute
+        execute
       end
```

Comment:
I would probably move this logic into https://github.com/Katello/katello/blob/master/app/lib/actions/pulp/repository/sync.rb#L60-L73  for a couple reasons:

a) Currently ~5 actions are using this middleware, so it means we're reaching out to pulp 5 times for the same information.  This is minor though.
b)  Right now since the IndexContent step is first this properly detects that the index needs to happen.  Then the other actions occur and get skipped because the indexed content now matches what is in pulp.  We likely need to execute those actions though as they likely were not executed previously.